redact/redact.go

```diff
@@ -110,6 +114,82 @@ func Bytes(b []byte, redactions []*Redact) ([]byte, error) {
 	return buf.Bytes(), nil
 }
 
+// JSON accepts a json map or array and traverses the collections and redacts any strings we find.
+func JSON(a any, redactions []*Redact) (any, error) {
+	switch coll := a.(type) {
+	case map[string]any:
+		r, err := redactMap(coll, redactions)
+		if err != nil {
+			return nil, err
+		}
+		return r, nil
+	case []any:
+		r, err := redactArray(coll, redactions)
+		if err != nil {
+			return nil, err
+		}
+		return r, nil
+	default:
+		return nil, fmt.Errorf("json collection to redact is neither a map nor an array, coll=%v", coll)
+	}
+}
+
+func redactArray(a []any, redactions []*Redact) ([]any, error) {
```

Comment:
Sorry to be pedantic here, but should this be `redactSlice` actually?

Reply:
Might as well! We're a few steps past json.unmarshall at this point so it's not really a json array anymore.